src/parser.ml

```diff
@@ -303,15 +303,17 @@ let parser arg_ident =
 
 (** Metavariable identifier (regular or escaped, prefixed with ['?']). *)
 let parser meta =
-  | "?" - id:{regular_ident | escaped_ident} -> in_pos _loc id
+  | "?" - id:{regular_ident | escaped_ident} ->
+      if id = "_" then Earley.give_up (); in_pos _loc id
```

Comment:
Will this generate a parse error? If so, couldn't you provide an informative error message? Perhaps there are other calls to Early.give_up() where we could do this as well.

Reply:
Yes, but what you ask is not easily possible with Earley. I will probably replace Earley in the near future to solve that problem and the complexity problem as well.